src/graph.erl

```diff
@@ -0,0 +1,270 @@
+%%    Response = persistence:set(Id, Json),
+%%    case Response of
+%%        {error, Reason} -> {error, Reason};
+%%        {ok, _Result} -> ok
+%%    end.
+%%
+%%delete(Id) ->
+%%    Response = persistence:del(Id),
+%%    case Response of
+%%        {error, Reason} -> {error, Reason};
+%%        {ok, _Result} -> ok
+%%    end.
+%%
+%%children_key(Key) when is_binary(Key) == true ->
+%%    <<Key/binary, "/children">>;
+%%children_key(Key) -> children_key(binary:list_to_bin(Key)).
+%%
+%%add_user_to_group(User, Group) ->
+%%    persistence:set_add(children_key(Group), User).
+%%
+%%remove_user_from_group(User, Group) ->
+%%    persistence:set_remove(children_key(Group), User).
+%%
+%%is_member_of_group(User, Group) ->
+%%    {ok, Result} = persistence:set_is_member(children_key(Group), User),
+%%    Result.
+%%
+%%list_group_users(Group) ->
+%%    {ok, Result} = persistence:set_list_members(children_key(Group)),
+%%    Result.
+%%
+%%
+%%add_group(Name) ->
+%%    Id = generate_id(),
+%%    Json = json_utils:encode(#{<<"type">> => <<"group">>, <<"id">> => Id, <<"name">> => Name}),
+%%    Response = persistence:set(Id, Json),
+%%    case Response of
+%%        {error, Reason} -> {error, Reason};
+%%        {ok, _Result} -> {ok, Id}
+%%    end.
+%%
+%%
+%%
+%%create_edge(Parent, Child, Permissions) ->
+%%    Flag = case {vertex_exists(Parent), vertex_exists(Child)} of
+%%               {false, _} -> false;
+%%               {_, false} -> false;
+%%               _ -> true
+%%           end,
+%%    case Flag of
+%%        false -> {error, "At least one of nodes doesn't exist"};
+%%        true ->
+%%            persistence:add_parent(Child, Parent, Permissions),
+%%            persistence:add_child(Parent, Child, Permissions),
+%%            ok
+%%    end.
+%%
+%%vertex_exists(Key) ->
+%%    case persistence:if_exists(Key) of
+%%        {ok, 0} -> false;
+%%        {ok, 1} -> true;
+%%        _ -> error
+%%    end.
+
+
+%%%% Parent, Child, Vertex to wszystko ID-ki, jesli wolisz mozesz zmienic nazwy na ParentId itd., jak uwazasz
+
+-spec create_edge(Parent::binary(), Child::binary(), Permissions::binary()) -> ok | {error, any()}.
+create_edge(_Arg0, _Arg1, _Arg2) ->
+    erlang:error(not_implemented).
+
+-spec update_edge(Parent::binary(), Child::binary(), Permissions::binary()) -> ok | {error, any()}.
+update_edge(_Arg0, _Arg1, _Arg2) ->
+    erlang:error(not_implemented).
+
+-spec remove_edge(Parent::binary(), Child::binary()) -> ok | {error, any()}.
+remove_edge(_Arg0, _Arg1) ->
+    erlang:error(not_implemented).
+
+-spec edge_exists(Parent::binary(), Child::binary()) -> true | false | {error, any()}.
+edge_exists(_Arg0, _Arg1) ->
+    erlang:error(not_implemented).
+
+-spec get_edge(Parent::binary(), Child::binary()) -> {ok, map()} | {error, any()}.
+get_edge(_Arg0, _Arg1) ->
+    erlang:error(not_implemented).
+
+%% @todo pierwszy klucz w mapie to <<"parents">>, a drugi to <<"children">>, ale nie wiem jak to przekazac w -spec
```

Comment:
Też nie wiem i nie znalazłem nic w internecie na ten temat, chyba się po prostu nie da. Czy tak nie było by czytelniej?
```suggestion
%% %{ <<"parents">> => binary(), <<"children">> => binary()}
```

Reply:
Jest czytelniej, ale chciałem przekazać, żebyś to zrobił jeśli umiesz